src/applications/lifecycles/mount.js

```diff
@@ -21,7 +21,12 @@ export async function toMountPromise(appOrParcel) {
     await reasonableTime(appOrParcel.mount(getProps(appOrParcel)), `Mounting application '${appOrParcel.name}'`, appOrParcel.timeouts.mount);
     appOrParcel.status = MOUNTED;
   } catch (err) {
-    handleAppError(err, appOrParcel);
+    if (!hardFail) {
+      handleAppError(err, appOrParcel);
+    } else {
+      const transformedErr = transformErr(err, appOrParcel)
+      throw transformedErr
+    }
     appOrParcel.status = SKIP_BECAUSE_BROKEN;
```

Comment:
I think this is a bug -- this will never get executed if line 28 is executed. Don't we want to change the status of the appOrParcel, no matter what?

Reply:
Good catch. I'll fix that